src/main/webapp/script.js

```diff
@@ -12,22 +23,122 @@ function listElections() {
+          }
+        } else {
+          nationalElections.append(election);
         }
       });
       
-      buildElectionListElements(stateElections);
+      const electionListContainerElement = document.getElementById('election-list-content');
+      
+      const electionsInStateHeadingElement = document.createElement('h3');
+      electionsInStateHeadingElement.innerHTML = 'Elections coming up in ' + selectedStateName;
+      electionListContainerElement.appendChild(electionsInStateHeadingElement);
+
+      if (stateElections.length == 0) {
+        buildElectionListElements(electionListContainerElement, stateElections);
+      } else {
+        buildEmptyListText(electionListContainerElement);
+      }
+
+      const electionsInNationHeadingElement = document.createElement('h3');
+      electionsInNationHeadingElement.innerHTML = 'Elections coming up in the United States';
+      electionListContainerElement.appendChild(electionsInNationHeadingElement);
+
+      if (nationalElections.length == 0) {
+        buildElectionListElements(electionListContainerElement, nationalElections);
+      } else {
+        buildEmptyListText(electionListContainerElement);
+      }
+  });
+}
+
+/**
+ * 
+ * @param {*} containerElement 
+ * @param {*} elections 
+ */
+function buildElectionListElements(containerElement, elections) {
+  const electionListElement = document.createElement('ul');
+  electionListElement.className = 'list-group list-group-flush';
+  electionListElement.id = 'election-list';
+
+  elections.forEach((election) => {
+    const electionItemElement = document.createElement('li');
+    electionItemElement.className = 'list-group-item';
+    electionItemElement.id = 'election-item';
+
+    const electionDetailsElement = document.createElement('div');
+    electionDetailsElement.className = 'election-quick-details';
+
+    const electionNameElement = document.createElement('p');
+    electionNameElement.innerHTML = election.name;
+    electionDetailsElement.appendChild(electionNameElement);
+
+    const electionDateElement = document.createElement('p');
+    const electionDateEmphasisElement = document.createElement('em');
+    electionDateEmphasisElement.innerHTML = formatDate(election.date);
+    electionDateElement.appendChild(electionDateEmphasisElement);
+    electionDetailsElement.appendChild(electionDateElement);
+
+    electionItemElement.appendChild(electionDetailsElement);
+
+    const learnMoreButtonElement = document.createElement('div');
+    learnMoreButtonElement.className = 'learn-more-button';
+    learnMoreButtonElement.innerHTML = 'Learn more';
+
+    electionItemElement.appendChild(learnMoreButtonElement);
+
+    electionListElement.appendChild(electionItemElement);
   });
+
+  containerElement.appendChild(electionListElement);
 }
 
-function buildElectionListElements(elections) {
-  const electionListContainerElement = document.getElementById('election-list-content');
+/**
+ * Build the text that 
```

Comment:
That what?

Reply:
Haha, again old commit where I hadn't finished documenting. Should I be consistently sticking documentation & implementation in the same commit?

Reply:
hm.. I think in terms of "git" commits, it probably doesn't matter. But the documentation & implementation should happen in the same pull request if that makes sense.

Reply:
Yeah that's fine. Sorry, github just makes it hard to review since I need to go through all the commit, but some of the commits overwrite each other.

Reply:
I usually look it over through "Files changed", I think that has the most up-to-date code